contracts/contracts/RecoveryWithDelayPlugin.sol

```diff
@@ -0,0 +1,281 @@
+
+        SafeProtocolAction memory safeProtocolAction = SafeProtocolAction(payable(address(safe)), 0, txData);
+        SafeRootAccess memory safeTx = SafeRootAccess(safeProtocolAction, 0, "");
+        (data) = manager.executeRootAccess(safe, safeTx);
+
+        emit OwnerReplaced(address(safe), oldOwner, newOwner);
+    }
+
+    /**
+     * @dev Returns the chain id used by this contract.
+     */
+    function getChainId() public view returns (uint256) {
+        uint256 id;
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            id := chainid()
+        }
+        return id;
+    }
+
+    /**
+     * @notice Returns the transaction hash for a recovery transaction.
+     * @param manager Address of the manager contract.
+     * @param account Address of the safe account.
+     * @param prevOwner Address of the owner previous to the owner to be replaced in the linked list
+     * @param oldOwner Address of the owner to be replaced.
+     * @param newOwner Address of the new owner.
+     * @param nonce A uint256 used to uniquely identify a recovery transaction.
+     */
+    function getTransactionHashData(
+        address manager,
+        address account,
+        address prevOwner,
+        address oldOwner,
+        address newOwner,
+        uint256 nonce
+    ) public view returns (bytes memory) {
+        uint256 chainId = getChainId();
+
+        bytes32 domainSeparator = keccak256(abi.encode(DOMAIN_SEPARATOR_TYPEHASH, chainId, this));
+
+        bytes32 transactionHash = keccak256(
+            abi.encode(DELAYED_RECOVERY_TRANSACTION_TYPEHASH, recoverer, manager, account, prevOwner, oldOwner, newOwner, nonce)
+        );
+
+        return abi.encodePacked(bytes1(0x19), bytes1(0x01), domainSeparator, transactionHash);
+    }
+
+    /**
+     * @notice Creates a recovery announcement for a Safe account. Only the recoverer can create a recovery announcement.
+     * @param manager Address of the manager contract.
+     * @param account Address of the safe account.
+     * @param prevOwner Address of the owner previous to the owner to be replaced in the linked list
+     * @param oldOwner Address of the owner to be replaced.
+     * @param newOwner Address of the new owner.
+     * @param nonce A uint256 used to uniquely identify a recovery transaction.
+     * @param executionTime A uint64 representing the block time in seconds after which the announced transaction can be executed.
+     */
+    function createAnnouncement(
+        address manager,
+        address account,
+        address prevOwner,
+        address oldOwner,
+        address newOwner,
+        uint256 nonce,
+        uint64 executionTime,
+        uint64 validityDuration
+    ) external onlyRecoverer {
+        bytes32 txHash = getTransactionHash(manager, account, prevOwner, oldOwner, newOwner, nonce);
+        Announcement memory announcement = announcements[txHash];
```

Comment:
does this write to storagE?

Reply:
line `182` is intended to only read from storage.